x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/metadata/add_metadata_filter_button.tsx

```diff
@@ -0,0 +1,123 @@
+interface AddMetadataFilterButtonProps {
+  item: {
+    name: string;
+    value: string | string[] | undefined;
+  };
+}
+
+const filterAddedToastTitle = i18n.translate(
+  'xpack.infra.hostsViewPage.flyout.metadata.filterAdded',
+  {
+    defaultMessage: 'Filter was added',
+  }
+);
+
+export const AddMetadataFilterButton = ({ item }: AddMetadataFilterButtonProps) => {
+  const { dataView } = useMetricsDataViewContext();
+  const { searchCriteria } = useUnifiedSearchContext();
+  const [existingFilter, setExistingFilter] = useState<
+    PhrasesFilter | PhraseFilter | ScriptedPhraseFilter | Filter | undefined
+  >(() => undefined);
+  const {
+    services: {
+      data: {
+        query: { filterManager: filterManagerService },
+      },
+      notifications: { toasts: toastsService },
+    },
+  } = useKibanaContextForPlugin();
+
+  useMemo(() => {
+    const addedFilter = searchCriteria.filters.find((filter) => filter.meta.key === item.name);
+    if (addedFilter) {
+      setExistingFilter(addedFilter);
+    }
+  }, [item.name, searchCriteria.filters]);
+
+  if (existingFilter) {
+    return (
+      <span>
+        <EuiToolTip
+          content={i18n.translate(
+            'xpack.infra.hostsViewPage.flyout.metadata.setRemoveFilterTooltip',
+            {
+              defaultMessage: 'Remove filter',
+            }
+          )}
+        >
+          <EuiButtonIcon
+            size="s"
+            color="text"
+            iconType="filter"
+            display="base"
+            data-test-subj="hostsView-flyout-metadata-remove-filter"
+            aria-label={i18n.translate(
+              'xpack.infra.hostsViewPage.flyout.metadata.filterAriaLabel',
+              {
+                defaultMessage: 'Filter',
+              }
+            )}
+            onClick={() => filterManagerService.removeFilter(existingFilter)}
+          />
+        </EuiToolTip>
+      </span>
+    );
+  }
+
+  return (
+    <span>
+      <EuiToolTip
+        content={i18n.translate(
+          'xpack.infra.hostsViewPage.flyout.metadata.setFilterByValueTooltip',
+          {
+            defaultMessage: 'Filter by value',
+          }
+        )}
+      >
+        <EuiButtonIcon
+          color="primary"
+          size="s"
+          iconType="filter"
+          data-test-subj="hostsView-flyout-metadata-add-filter"
+          aria-label={i18n.translate(
+            'xpack.infra.hostsViewPage.flyout.metadata.AddFilterAriaLabel',
+            {
+              defaultMessage: 'Add Filter',
+            }
+          )}
+          onClick={() => {
+            const newFilter = buildMetadataFilter({
+              field: item.name,
+              value: item.value ?? '',
+              dataView,
+              negate: false,
+            });
+            if (newFilter) {
+              filterManagerService.addFilters(newFilter);
+              toastsService.addSuccess({
+                title: filterAddedToastTitle,
+                toastLifeTimeMs: 10000,
```

Comment:
**question**: are 10 seconds the required time for the notification? It seems quite long, we usually set it to 3 seconds

Reply:
I tested with different intervals and if it's the 3-second one (or less) sometimes the toast disappears while the page is still loading. I showed it to @roshan-elastic  to verify that it's Ok to stay so the user will see it after the page loads - it can be closed in case it's annoying